AutoFilterEngine implementing open

magnusuMET opened the pull request:
The `AutoFilterEngine` should for parity implement `open` as `args()` inspects `__init__` of `reader_class`

magnusuMET committed:
AutoFilterEngine implementing open

heikoklein reviewed (changes requested):
Thanks, this is a welcome simplification for the AutoFilterReaderEngine.

Please update the documenation accordingly in https://github.com/metno/pyaro/blob/1db597e91b8e9cc281d1f4df4ca765eb93a3f87d/docs/how-to-add-new-reader.rst?plain=1#L35

The support of type-checking (py.typed) can be discussed, but it should clearly not be a part of this PR, but a PR of it's own.

magnusuMET committed:
Update docs on AutoFilterEngine

magnusuMET commented:
Thanks for the review. I updated the docs on how to implement `AutoFilterReaderEngine`. I additionally fixed `filters=None`, as we always assume `filters` is an iterable.

`py.typed` was an accidental inclusion, removed this.

heikoklein commented on docs/how-to-add-new-reader.rst:
```diff
@@ -32,8 +32,8 @@ Subclassing of Engine/Reader using AutoFilterReaderEngine
 
-- :py:meth:`~pyaro.timeseries.Engine.open`: opening a reader with the signature (self, filename, *args, **kwargs) -> YourReader``,
-  usually just like: ``return YourReader(filename, *args, **kwargs)``
+- :py:meth:`~pyaro.timeseries.AutoFilterReader.reader_class`: a class implementing `AutoFilterReader`,
```
maybe this should be more specific:

the class implementing the `AutoFilterReader` corresponding to this `AutoFilterEnginge`, e.g. the reader_class() of `YourEngine` should return `YourReader`

magnusuMET committed:
Doc clarification
